### PR TITLE
Update create-dns-zones.sh

### PR DESCRIPTION
**Pull request by tcaky**
moved to array-based loop because of @simardeep1792 suggestion.

### CODE REVIEW

**Comment by ericmalen** [comment]
LGTM. I have one question though. Is there a reason we're using an indexed array here instead of an associative array? IMO using an associative array would be more appropriate. Not trying to be nit-picky, I'm genuinely curious as I don't have much experience writing Bash scripts, so I'm wondering if there's something I don't know or if it boils down to personal preference.

@simardeep1792 - Tagging you because this was originally your suggestion.

File: infra-deployment-scripts/create-dns-zones.sh [hunk 1]
```diff
@@ -1,30 +1,42 @@
...
+    "open-ouvert-phac-aspc-gc-ca:open-ouvert.phac-aspc.gc.ca."
+)
 
-for dnsName in "${dnsNames[@]}"; do
-    zoneName="${dnsName//./-}" # Replace dots with dashes for the zone name
-    gcloud dns --project "${GOOGLE_CLOUD_PROJECT}" managed-zones create "${zoneName}" \
-        --description="" --dns-name="${dnsName}." --visibility="public" \
-        --dnssec-state="on" --log-dns-queries
+# Loop through the array and create each managed zone
+for zone_info in "${zones[@]}"; do
```
Comment (ericmalen) [inline infra-deployment-scripts/create-dns-zones.sh:39]: for zone_name in "${!zones[@]}"; do
    dns_name=${zones[$zone_name]}
    gcloud dns --project "${GOOGLE_CLOUD_PROJECT}" managed-zones create "$zone_name" --description="" --dns-name="$dns_name" --visibility="public" --dnssec-state="on" --log-dns-queries
done

**Comment by tcaky** [comment]
> LGTM. I have one question though. Is there a reason we're using an indexed array here instead of an associative array? IMO using an associative array would be more appropriate. Not trying to be nit-picky, I'm genuinely curious as I don't have much experience writing Bash scripts, so I'm wondering if there's something I don't know or if it boils down to personal preference.
> 
> @simardeep1792 - Tagging you because this was originally your suggestion.

It was just what Simardeep had submitted.  I wasn't married to anything TBH.  If I was writing it myself, I likely would have used a hashmap/associative array as well.

**Comment by tcaky** [comment]
@ericmalen  actually the script is likely going to be deleted soon, the new PR I put in to describe all the zones in YAML is the first step in making config-controller responsible for the zones - at which point there is no need to keep the script. :)

**Comment by ericmalen** [comment]
> > LGTM. I have one question though. Is there a reason we're using an indexed array here instead of an associative array? IMO using an associative array would be more appropriate. Not trying to be nit-picky, I'm genuinely curious as I don't have much experience writing Bash scripts, so I'm wondering if there's something I don't know or if it boils down to personal preference.
> > @simardeep1792 - Tagging you because this was originally your suggestion.
> 
> It was just what Simardeep had submitted. I wasn't married to anything TBH. If I was writing it myself, I likely would have used a hashmap/associative array as well.

> @ericmalen actually the script is likely going to be deleted soon, the new PR I put in to describe all the zones in YAML is the first step in making config-controller responsible for the zones - at which point there is no need to keep the script. :)

Cool, just wanted to make sure there wasn't anything I should be aware of. Thanks!

**Review by ericmalen — APPROVED** [review]
👍

**Pull request merged**